packages/core/src/Cardano/types/ProtocolParameters.ts

```diff
@@ -6,9 +6,12 @@ export interface ProtocolVersion {
   minor: number;
   patch?: number;
 }
-export type CostModels = CostModel[];
-
-export type CostModel = number[];
+export interface CostModels {
+  [k: string]: CostModel;
```

Comment:
Please add a type aliases for all primitive types (string & number). It is not clear what they are.

Reply:
There are currently two values this could be `PlutusV1` or `PlutusV2`

Reply:
I suggest refactoring [this enum](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/core/src/Cardano/types/Script.ts#L193) to have string values and doing this:
```suggestion
  [k in PlutusLanguageVersion]: CostModel | undefined;
```

@AngelCastilloB would that be ok or are number values of that enum important?

Reply:
The values are not important, we can change it to strings without problem. However I think this:

` [k: string]: CostModel;`

Should be typed as this:

`export type CostModels = CostModel[]`

The `[k: string]: Value` is the way Ogmios represent arrays, since arrays in the CDDL are represented as maps where the keys is the numerical position (index) in the array.

Reply:
From CDDL:
```
costmdls =
  { ? 0 : [ 166*166 int ] ; Plutus v1
  , ? 1 : [ 175*175 int ] ; Plutus v2
  }
```

something like:

```
models[0] = plutusV1costmdls;
models[1] = plutusV2costmdls;
```

Reply:
Thanks @AngelCastilloB. Agree, let's go with `CostModel[]`. Perhaps it would then make sense to update the enum to correspond to cost model index, so that you could do: `costModels[PlutusLanguageVersion.PlutusV1]`?

Reply:
Yes @mkazlauskas :), I am not sure why I initialized the enum with 1 instead of 0, that was pretty random

Reply:
How about?:
```ts
export enum PlutusLanguageVersion {
  V1 = 'PlutusV1',
  V2 = 'PlutusV2'
}
```

Reply:
Following all suggestions, do we all agree with:

1. Fix [PlutusLanguageVersion](https://github.com/input-output-hk/cardano-js-sdk/blob/3b757ba77e163b44136227ae2a8fa59600b4474f/packages/core/src/Cardano/types/Script.ts#L189-L207) enum to
`export enum PlutusLanguageVersion {
  V1 = 0,
  V2 = 1
}`

2. Moving cost model types to array:

   `export type CostModel = {
  [k: string]: number;
}`

    `export type CostModels = CostModel[]`

3. And then would have:

```
models: CostModels;

models[PlutusLanguageVersion.V1] = plutusV1costModel;
models[PlutusLanguageVersion.V2] = plutusV2costModel;
```

@rhyslbw @mkazlauskas @AngelCastilloB

Reply:
The changes are applied here 004f780300900891ba24d193125644e6c31f8d30

Only to note, regarding the introduced `PlutusLanguageVersion` enum fix and the keys renaming to `V1` and `V2`, I checked that in `cardano-services` config genesis files remain as:
```
"costModels": {
        "PlutusV1": {
            "sha2_256-memory-arguments": 4,
            ...
            }
         }
```

Reply:
We'll just need a translation between the object and array representation.